# Data description

## Activate and inactive chromosome X (.summary.txt.gz files) from Wang *et al.*

Data generated by [Wang *et al.* 2019](https://pubmed.ncbi.nlm.nih.gov/29887375/) and featured in [Lappala *et al.* 2021](https://www.pnas.org/doi/abs/10.1073/pnas.2107092118) were downloaded from the [Gene Expression Omnibus](https://www.ncbi.nlm.nih.gov/geo/) in the .summary.txt.gz file format. The first wild-type replicate from the record [GSE99991](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE99991) with accession record [GSM2667262](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2667262) was used in analysis and development of the *.summary.txt.gz* to *.hic* executable script ([listed here](https://github.com/4DGB/hic-converter/blob/main/tools/summary.to.chrom.hic.py)). Here we store the two haplotype resolved *.summary.txt.gz* files for chromosome 13 of the first replicate from Wang *et al.* for the wild-type, cas and mus haplotypes: 

    - /summary/GSM2667262_WT1.HiC.rep1.cas.chr13.summary.txt.gz 
    - /summary/GSM2667262_WT1.HiC.rep1.mus.chr13.summary.txt.gz

The *cas* and *mus* haplotype names correspond to the activate (Xa) and inactive (Xi) chromosome X states, respectively. The mm9, Mus musculus reference genome was used in analysis.

### Generating a *.hic* file from *.summary.txt.gz* file
```
## Change directory to tools
cd ./hic-converter/tools

## Convert a summary.txt.gz file to an .hic file for a single chromosome
./summary.to.chrom.hic.py -i ../data/summary/GSM2667262_WT1.HiC.rep1.cas.chr13.summary.txt.gz -g mm9 -c chr13 -O ../data/hic/GSM2667262_WT1.HiC.rep1.cas.chr13.hic
```

## Hi-C data on mouse embryonic stem cells from Marks *et al.* 

The following code and instructions were modified from a [HicExplorer tutorial]( https://hicexplorer.readthedocs.io/en/latest/content/mES-HiC_analysis.html) on aligning Hi-C data from female mouse, embryonic stem cells. It was adapted to generate a Hi-C contact map for chromosome 13 in *.h5* format. Additionally only data from the first replicate of [Marks *et al.* 2015]( https://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0698-x) is used in this analysis. See the [main page]( https://github.com/4DGB/hic-converter) of this repository for instruction on setting up a python computing environment and converting the *.h5* file to a *.hic* file.

### Constructing a *.h5* contact map of chromosome 13 from pair-end *.fastq.gz* files

```
## Set the working directory
cd ./hic-converter/data

## Activate the python computing environment.
conda activate hicexplorerenv

## Make directories for analysis.
mkdir -p genome_mm10 fastq bam bwa

## Gather the mm10 reference, use wget and tar to unpack the data.
wget http://hgdownload-test.cse.ucsc.edu/goldenPath/mm10/bigZips/chromFa.tar.gz -O genome_mm10/chromFa.tar.gz
tar -xvzf genome_mm10/chromFa.tar.gz

## Index chromosome 13 via bwa index.
bwa index -p bwa/chr13_index genome_mm10/chr13.fa

## Use the hicFindRestSite function from HicExplorer to find the SauIII restriction sites (GATC) across the mm10 genome. 
hicFindRestSite --fasta ./genome_mm10/chr13.fa -o ./genome_mm10/chr13_SauIII_cut_sites.bed --searchPattern GATC

## Download data from Marks et al. for alignment.
wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR195/007/SRR1956527/SRR1956527_1.fastq.gz -O ./fastq/SRR1956527_1.fastq.gz
wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR195/007/SRR1956527/SRR1956527_2.fastq.gz -O ./fastq/SRR1956527_2.fastq.gz

## Align data seperately for each fastq file to chromosome 13 via bwa mem. Note we do not sort or filter the output bam files.
bwa mem -A 1 -B 4 -E 50 -L 0 -t 14 bwa/chr13_index ./fastq/SRR1956527_1.fastq.gz | samtools view -Shb - > ./bam/SRR1956527_1_chr13.bam
bwa mem -A 1 -B 4 -E 50 -L 0 -t 14 bwa/chr13_index ./fastq/SRR1956527_2.fastq.gz | samtools view -Shb - > ./bam/SRR1956527_2_chr13.bam

## Construct Hi-C contact matrix in .h5 format via the build matrix function from HicExplorer.
hicBuildMatrix --samFiles ./bam/SRR1956527_1_chr13.bam ./bam/SRR1956527_2_chr13.bam \
    --restrictionSequence GATC --danglingSequence GATC --restrictionCutFile ./genome_mm10/chr13_SauIII_cut_sites.bed \
    --threads 5 --inputBufferSize 400000 \
    --QCfolder ./SRR1956527_chr13_QC --outFileName ./h5/SRR1956527_chr13.h5
```

## A549 Cells From the ENCODE Project

The data listed here was generated for the ENCODE project by the Reddy lab at Duke University. It includes paired-end, Hi-C sequencing libraries on A549 cells, a cancer lung cell line treated with 100 nM dexamethasone. Specifically, the library of the first isogenic replicate ([ENCLB571GEP](https://www.encodeproject.org/experiments/ENCSR662QKG/)) with paired files [ENCFF039FYU](https://www.encodeproject.org/files/ENCFF039FYU/) and [ENCFF479RSE](https://www.encodeproject.org/files/ENCFF479RSE/) for the initial time point and the first isogenic replicate ([ENCLB870JCZ](https://www.encodeproject.org/experiments/ENCSR499RVD/)) with paired files [ENCFF668EDF](https://www.encodeproject.org/files/ENCFF668EDF/) and [ENCFF398SQH](https://www.encodeproject.org/files/ENCFF398SQH/) for the twelve hour time point (treated with 100 nM dexamethasone) were used to generate hic files via HiCExplorer. As an example, we have included the instructions for the construciton of the .h5 file of the twelve hour timepoint below.

### Building a contact map for chromosome 22

```
## Set the working directory
cd ./hic-converter/data

## Make directories for analysis.
mkdir -p GRCh38 fastq bam bwa

## Download the two paired fastq.gz files for the first replicate.
wget https://www.encodeproject.org/files/ENCFF668EDF/@@download/ENCFF668EDF.fastq.gz -O ./fastq/ENCFF668EDF_R1_001.fastq.gz
wget https://www.encodeproject.org/files/ENCFF398SQH/@@download/ENCFF398SQH.fastq.gz -O ./fastq/ENCFF398SQH_R2_001.fastq.gz

## Download the human reference genome from the ENCODE project.
wget https://www.encodeproject.org/files/GRCh38_no_alt_analysis_set_GCA_000001405.15/@@download/GRCh38_no_alt_analysis_set_GCA_000001405.15.fasta.gz -O ./GRCh38/GRCh38_no_alt_analysis_set_GCA_000001405.15.fasta.gz

## Unzip the genome.
gunzip ./GRCh38/GRCh38_no_alt_analysis_set_GCA_000001405.15.fasta.gz

## Find the MobI restriction sites (GATC) in the GRCh38, human reference genome.
hicFindRestSite --fasta ./GRCh38/GRCh38_no_alt_analysis_set_GCA_000001405.15.fasta -o ./GRCh38/GRCh38_MobI_cut_sites.bed --searchPattern GATC

## Index the human genome (this can take awhile but we only need to do it once)!
bwa index -p bwa/GRCh38_index ./GRCh38/GRCh38_no_alt_analysis_set_GCA_000001405.15.fasta 

## Align data to the human reference genome. 
bwa mem -A 1 -B 4 -E 50 -L 0 -t 14 bwa/GRCh38_index ./fastq/ENCFF668EDF_R1_001.fastq.gz | samtools view -SHb - > ./bam/ENCFF668EDF_R1_001.bam
bwa mem -A 1 -B 4 -E 50 -L 0 -t 14 bwa/GRCh38_index ./fastq/ENCFF398SQH_R2_001.fastq.gz | samtools view -SHb - > ./bam/ENCFF398SQH_R2_001.bam

## Build .h5 matrix for chromosome 22.
hicBuildMatrix --samFiles ./bam/ENCFF668EDF_R1_001.bam ./bam/ENCFF398SQH_R2_001.bam \
    --region chr22 --binSize 200000 \
    --threads 4 --inputBufferSize 200000 \
    --restrictionSequence GATC --danglingSequence GATC --restrictionCutFile ./GRCh38/GRCh38_MboI_cut_sites.bed \
    --QCfolder ENCLB870JCZ.chr22.12hr --outFileName ./h5/ENCLB870JCZ.chr22.200kb.12hr.h5
```

### Converting the chromosome 22 *.h5* contact map to *.hic*

```
## Change directory to tools
cd ./hic-converter/tools

## Convert a summary.txt.gz file to an .hic file for a single chromosome
./h5.to.hic.sh -m ../data/h5/ENCLB870JCZ.chr22.200kb.12hr.h5 -g ..data/sizes/GRCh38.chr22.size.bed -o ../data/hic/ENCLB870JCZ.chr22.200kb.12hr.hic
```